Vikta-API-testing-sample/src/test/java/com/test/api/category/CategoryTest.java

```diff
@@ -62,6 +79,95 @@ public void checkCategoryInAllCategoriesList() {
                 hasItem(equalTo(expectedCategory)));
     }
 
+
+
+    /**
+     *  Test with entity search
+     */
+    // it's weird, when extract.as(CategoryDTO[].class) return empty array, at http://localhost:5054/swagger-ui.html it works
+    // when extract.response the status code equals 200
+    @Test
+    public void checkIfCanFindEntity(){
+        final String termQueryParam = "term";
+        final String term = "Wolf%20N8%20%2F%20Dynamic%20Marketing%20Officer";
+
+        CategoryDTO[] city = given().spec(defaultRequestSpec())
+                .queryParam(termQueryParam, term)
+                .when()
+                .log().all()
+                .get(getTestEnvironment().getCategoryPath() + "/search")
+                .then()
+                .log().all()
+                .spec(defaultResponseJsonSpec())
+                .extract()
+                .as(CategoryDTO[].class);
+
+        assertThat(city[0].getTitle()).isEqualTo("City");
+    }
+
+    /**
+     *  Test with posting new category
+     */
+    // Intellij returns status code = 500
+    // "Could not commit JPA transaction; nested exception is javax.persistence.RollbackException: Error while committing the transaction"
+    // When using queryParam instead of CategoryDTO objects, returns code = 400 - same in postman
+
+    @Test
+    public void tryPostNewCategory(){
```

Comment:
you need to set all the fields in CategoryDTO (except for id, which is set automatically) and it will work